### PR TITLE
Task: Added a submission consent switch to the warn others notification DM,…

### PR DESCRIPTION
**Pull request by SwissMountainFrog**
Small PR which brings the submission consent switch to the Warn Others Notifications Developer Menu to make certain testings easier.

Based on [JIRA Ticket 3993](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-3993)

image

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Developer Menu/Features/DMWarnOthersNotificationViewController.swift [hunk 1]
```diff
@@ -58,7 +70,7 @@ final class DMWarnOthersNotificationViewController: UIViewController, UITextFiel
 		let currentSubmissionConsentStatusStateLabel = UILabel(frame: .zero)
 		currentSubmissionConsentStatusStateLabel.translatesAutoresizingMaskIntoConstraints = false
 		currentSubmissionConsentStatusStateLabel.numberOfLines = 0
-		currentSubmissionConsentStatusStateLabel.text = store.isSubmissionConsentGiven ? "🟢 Consent granted 👍" : "🔴 Consent not given 👎"
+		currentSubmissionConsentStatusStateLabel.text = ""
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Developer Menu/Features/DMWarnOthersNotificationViewController.swift:73]: I think this line can be removed completely.

File: src/xcode/ENA/ENA/Source/Developer Menu/Features/DMWarnOthersNotificationViewController.swift [hunk 2]
```diff
@@ -155,7 +178,23 @@ final class DMWarnOthersNotificationViewController: UIViewController, UITextFiel
...
+	
+	private var currentSubmissionConsentStatusTitleLabel = UILabel()
+	
+	@objc
+	private func consentStateChanged(switchState: UISwitch) {
+		exposureSubmissionService.setSubmissionConsentGiven(consentGiven: switchState.isOn)
+		updateConsentStatusLabel()
+	}
+	
+	private func updateConsentStatusLabel() {
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Developer Menu/Features/DMWarnOthersNotificationViewController.swift:190]: Can't this be removed and handled with the sink in the init?

**Review by marcussc — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Developer Menu/Features/DMWarnOthersNotificationViewController.swift [hunk 3]
```diff
@@ -155,7 +178,23 @@ final class DMWarnOthersNotificationViewController: UIViewController, UITextFiel
...
+		self.exposureSubmissionService.isSubmissionConsentGivenPublisher.sink { isSubmissionConsentGiven in
+			self.currentSubmissionConsentStatusTitleLabel.text = isSubmissionConsentGiven ? "🟢 Consent granted 👍" : "🔴 Consent not given 👎"
+		}.store(in: &cancellables)
+	}
+	
+	private var currentStatusLabelText = ""
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Developer Menu/Features/DMWarnOthersNotificationViewController.swift:196]: 🤔 what is this needed for?

File: src/xcode/ENA/ENA/Source/Developer Menu/Features/DMWarnOthersNotificationViewController.swift (hunk 1 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Developer Menu/Features/DMWarnOthersNotificationViewController.swift:73]: Sure. 👍  removed.

File: src/xcode/ENA/ENA/Source/Developer Menu/Features/DMWarnOthersNotificationViewController.swift (hunk 3 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Developer Menu/Features/DMWarnOthersNotificationViewController.swift:195]: No clue 😜  scrubbed out...

File: src/xcode/ENA/ENA/Source/Developer Menu/Features/DMWarnOthersNotificationViewController.swift (hunk 2 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Developer Menu/Features/DMWarnOthersNotificationViewController.swift:189]: Need to think on it for a moment 😉

Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Developer Menu/Features/DMWarnOthersNotificationViewController.swift:189]: The function is also used in `viewDidLoad` for initial rendering.

Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Developer Menu/Features/DMWarnOthersNotificationViewController.swift]: Completely refactored. Way more "combine" now.

**Review by marcussc — APPROVED** [review]

**Review by KaiTeuber — APPROVED** [review]

**Pull request merged**